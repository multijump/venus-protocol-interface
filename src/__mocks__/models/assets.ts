import { Asset } from 'types';

import governanceResponse from '__mocks__/api/governance.json';
import { formatToAsset } from 'clients/api/queries/getAssets';

export const assets: Asset[] = governanceResponse.data.markets
  .map(formatToAsset)
  .filter(asset => !!asset) as Asset[];
