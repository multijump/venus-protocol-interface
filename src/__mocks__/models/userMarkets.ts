import { UserMarket } from 'types';

import { userAssets } from './userAssets';

export const userMarkets: UserMarket[] = [
  {
    id: '1',
    riskLevel: 'MINIMAL',
    name: 'Venus',
    assets: userAssets,
  },
  {
    id: '2',
    riskLevel: 'VERY_HIGH',
    name: 'Gaming',
    assets: userAssets,
  },
  {
    id: '3',
    riskLevel: 'MEDIUM',
    name: 'MetaVerse',
    assets: userAssets,
  },
];
