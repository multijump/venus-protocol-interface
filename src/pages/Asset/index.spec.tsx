import { waitFor } from '@testing-library/react';
import BigNumber from 'bignumber.js';
import { createMemoryHistory } from 'history';
import React from 'react';

import { marketSnapshots } from '__mocks__/models/marketSnapshots';
import { markets } from '__mocks__/models/markets';
import { vTokenApySimulations } from '__mocks__/models/vTokenApySimulations';
import { getMarketHistory, getMarkets, getVTokenApySimulations } from 'clients/api';
import renderComponent from 'testUtils/renderComponent';

import UserMarket from '.';
import TEST_IDS from './testIds';

const fakeVTokenId = 'aave';
const fakeMarketId = 'fake-market-id';

jest.mock('clients/api');

describe('pages/UserMarket', () => {
  beforeEach(() => {
    (getMarketHistory as jest.Mock).mockImplementation(() => ({
      marketSnapshots,
    }));
    (getMarkets as jest.Mock).mockImplementation(() => ({
      markets,
      dailyXvsDistributedWei: new BigNumber(0),
    }));
    (getVTokenApySimulations as jest.Mock).mockImplementation(() => ({
      apySimulations: vTokenApySimulations,
    }));
  });

  it('renders without crashing', () => {
    const fakeHistory = createMemoryHistory();
    renderComponent(
      <UserMarket
        history={fakeHistory}
        location="/"
        match={{
          params: {
            vTokenId: fakeVTokenId,
            marketId: fakeMarketId,
          },
          isExact: true,
          path: '/:vTokenId',
          url: '',
        }}
      />,
    );
  });

  it('fetches market details and displays them correctly', async () => {
    const fakeHistory = createMemoryHistory();
    const { getByTestId } = renderComponent(
      <UserMarket
        history={fakeHistory}
        location="/"
        match={{
          params: {
            vTokenId: fakeVTokenId,
            marketId: fakeMarketId,
          },
          isExact: true,
          path: '/:vTokenId',
          url: '',
        }}
      />,
    );

    // Check supply info displays correctly
    await waitFor(() => expect(getByTestId(TEST_IDS.supplyInfo).textContent).toMatchSnapshot());
    // Check borrow info displays correctly
    expect(getByTestId(TEST_IDS.borrowInfo).textContent).toMatchSnapshot();
    // Check interest rate model displays correctly
    expect(getByTestId(TEST_IDS.interestRateModel).textContent).toMatchSnapshot();
    // Check market info displays correctly
    expect(getByTestId(TEST_IDS.marketInfo).textContent).toMatchSnapshot();
  });
});
