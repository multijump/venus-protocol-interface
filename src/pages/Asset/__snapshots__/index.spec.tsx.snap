// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Asset fetches market details and displays them correctly 1`] = `"Supply infoSupply APYTotal supply-APY-Distribution APY-"`;

exports[`pages/Asset fetches market details and displays them correctly 2`] = `"Borrow infoBorrow APYTotal borrow-APY-Distribution APY-"`;

exports[`pages/Asset fetches market details and displays them correctly 3`] = `"Interest Rate ModelUtilization rateBorrow APYSupply APY"`;

exports[`pages/Asset fetches market details and displays them correctly 4`] = `"Asset infoPrice$74.95Liquidity$811,477.26# of suppliers143# of borrowers18Borrow capUncappedDaily supplying interests$0.04Daily borrowing interests$0.04Daily XVS distributed25Reserves0 AAVEReserve factor0%Collateral factor0%vAAVE minted50,716,474,765,373.16458318Exchange rate1 AAVE=4641576581.659172 vAAVE"`;
